src/Chapter1.hs

```diff
@@ -187,474 +181,445 @@ ghci> :q
-
-Casual reminder about adding top-level type signatures for all functions :)
--}
-
-mid x y z = error "mid: not implemented!"
-
-{- |
-=⚔️= Task 8
-
-Implement a function that checks whether a given character is a vowel.
-
-🕯 HINT: use guards
-
->>> isVowel 'a'
-True
->>> isVowel 'x'
-False
--}
-isVowel c = error "isVowel: not implemented!"
-
-
-{- |
-== Local variables and functions
-
-So far, we've been playing only with simple expressions and function
-definitions. However, in some cases, expressions may become complicated, and it
-could make sense to introduce some helper variables.
-
-You can use the let-in construct in Haskell to define variables.
-Here goes an example:
-
-@
-half :: Int -> Int
-half n = let halfN = div n 2 in halfN
-@
-
-♫ NOTE: __let-in__ is also an expression! You can't just define variables; you
-  also need to return some expression that may use defined variables.
-
-The syntax for defining multiple variables requires to care about indentation
-more, but there is nothing special in it as well:
-
-@
-halfAndTwice :: Int -> (Int, Int)
-halfAndTwice n =
-    let halfN = div n 2
-        twiceN = n * 2
-    in (halfN, twiceN)
-@
-
-In addition to let-in (or sometimes even alternatively to let-in) you can use
-the __where__ construct to define local variables and functions.
-And, again, the example:
-
-@
-pythagoras :: Double -> Double -> Double
-pythagoras a b = square a + square b
+closestToZero x y = if abs x < abs y then x else y
+
+-- =⚔️= Task 7
+-- Write a function that returns the middle number among three given numbers.
+--
+-- >>> mid 3 1 2
+-- 2
+-- 🕯 HINT: When checking multiple conditions, it is more convenient to use the
+--   language construct called "guards" instead of multiple nested 'if-then-else'
+--   expressions. The syntax of guards is the following:
+-- @
+-- sign :: Int -> Int
+-- sign n
+--     | n < 0 = (-1)
+--     | n == 0 = 0
+--     | otherwise = 1
+-- @
+--
+-- You define different conditions in different branches, started by the '|'
+-- symbol. And the functions check them from top to bottom, returning the first
+-- value after "=" where the condition is true.
+--
+-- ♫ NOTE: The "=" sign goes after each branch, respectively.
+--
+-- ♫ NOTE: It is essential to have the same indentation before each branch "|"!
+--   Remember, that Haskell is an indentation- and layout-sensitive language. mid ::
+mid :: (Ord a) => a -> a -> a -> a
+mid x y z
+  | x > y && y > z = y
+  | y > z && z > x = z
+  | z > x && x > y = x
+  | x > z && z > y = z
+  | y > x && x > z = x
+  | z > y && y > x = y
+  | otherwise = error "invalid input"
+
+-- Casual reminder about adding top-level type signatures for all functions :)
+
+-- =⚔️= Task 8
+--
+-- Implement a function that checks whether a given character is a vowel.
+--
+-- 🕯 HINT: use guards
+--
+-- >>> isVowel 'a'
+-- True
+-- >>> isVowel 'x'
+-- False
+isVowel :: Char -> Bool
+isVowel c
+  | c `elem` "aeiouAEIOU" = True
+  | otherwise = False
```

Comment:
Nice one! 👍🏼 
As `elem` returns `Bool` itself, you can just return the result of this function:

```haskell
isVowel c = c `elem` "aeiouAEIOU"
```

One note in here, that sometimes, `elem` could be slower than the explicit pattern matching. I remember there were some benchmarks on one particular case, that showed how moving to pattern matching on each case separately drastically decrease time 🐎